src/ast/passes/semantic_analyser.cpp

```diff
@@ -3761,10 +3765,15 @@ void SemanticAnalyser::assign_map_type(const Map &map, const SizedType &type)
       else
         *maptype = type;
     } else if (maptype->GetTy() != type.GetTy()) {
+      std::stringstream hint;
+      if (maptype->IsIntegerTy() && type.IsCountTy() && expr.is_map)
```

Comment:
More than just count() right?

```suggestion
      if (maptype->IsIntegerTy() && type.IsCastableMapTy() && expr.is_map)
```

Reply:
Good call.